File: packages/java-sdk/src/main/java/com/turbodocx/models/GenerateTemplateResponse.java
package com.turbodocx.models;

import java.util.Map;

/**
 * Response from template generation
 */
public class GenerateTemplateResponse {
    private String id;
    private String name;
    private String description;
    private String templateId;
    private String projectspaceId;
    private String deliverableFolderId;
    private Map<String, Object> metadata;
    private String createdBy;
    private String orgId;
    private String defaultFont;
    private String createdOn;
    private String updatedOn;
    private Integer isActive;
    private Object fonts;
    private byte[] buffer;
    private String downloadUrl;
    private String message;
    private String error;

    // Constructors
    public GenerateTemplateResponse() {
    }

    // Getters and Setters
    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getTemplateId() {
        return templateId;
    }

    public void setTemplateId(String templateId) {
        this.templateId = templateId;
    }

    public String getProjectspaceId() {
        return projectspaceId;
    }

    public void setProjectspaceId(String projectspaceId) {
        this.projectspaceId = projectspaceId;
    }

    public String getDeliverableFolderId() {
        return deliverableFolderId;
    }

    public void setDeliverableFolderId(String deliverableFolderId) {
        this.deliverableFolderId = deliverableFolderId;
    }

    public Map<String, Object> getMetadata() {
        return metadata;
    }

    public void setMetadata(Map<String, Object> metadata) {
        this.metadata = metadata;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    public String getOrgId() {
        return orgId;
    }

    public void setOrgId(String orgId) {
        this.orgId = orgId;
    }

    public String getDefaultFont() {
        return defaultFont;
    }

    public void setDefaultFont(String defaultFont) {
        this.defaultFont = defaultFont;
    }

    public String getCreatedOn() {
        return createdOn;
    }

    public void setCreatedOn(String createdOn) {
        this.createdOn = createdOn;
    }

    public String getUpdatedOn() {
        return updatedOn;
    }

    public void setUpdatedOn(String updatedOn) {
        this.updatedOn = updatedOn;
    }

    public Integer getIsActive() {
        return isActive;
    }

    public void setIsActive(Integer isActive) {
        this.isActive = isActive;
    }

    public Object getFonts() {
        return fonts;
    }

    public void setFonts(Object fonts) {
        this.fonts = fonts;
    }

    public byte[] getBuffer() {
        return buffer;
    }

    public void setBuffer(byte[] buffer) {
        this.buffer = buffer;
    }

    public String getDownloadUrl() {
        return downloadUrl;
    }

    public void setDownloadUrl(String downloadUrl) {
        this.downloadUrl = downloadUrl;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    // Legacy compatibility - keep deliverableId as alias for id
    public String getDeliverableId() {
        return id;
    }

    public void setDeliverableId(String deliverableId) {
        this.id = deliverableId;
    }
}

File: packages/php-sdk/src/Types/Responses/GenerateTemplateResponse.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Responses;

/**
 * Response from generating a template
 */
class GenerateTemplateResponse
{
    /**
     * @param string|null $id Deliverable ID
     * @param string|null $name Document name
     * @param string|null $description Document description
     * @param string|null $templateId Template ID used
     * @param string|null $projectspaceId Projectspace ID
     * @param string|null $deliverableFolderId Folder ID
     * @param array<string, mixed>|null $metadata Additional metadata
     * @param string|null $createdBy User who created the deliverable
     * @param string|null $orgId Organization ID
     * @param string|null $defaultFont Default font used
     * @param string|null $createdOn Creation timestamp
     * @param string|null $updatedOn Last update timestamp
     * @param int|null $isActive Active status flag
     * @param mixed $fonts Font information
     * @param string|null $downloadUrl Download URL
     * @param string|null $message Response message
     * @param string|null $error Error message (if any)
     */
    public function __construct(
        public ?string $id = null,
        public ?string $name = null,
        public ?string $description = null,
        public ?string $templateId = null,
        public ?string $projectspaceId = null,
        public ?string $deliverableFolderId = null,
        public ?array $metadata = null,
        public ?string $createdBy = null,
        public ?string $orgId = null,
        public ?string $defaultFont = null,
        public ?string $createdOn = null,
        public ?string $updatedOn = null,
        public ?int $isActive = null,
        public mixed $fonts = null,
        public ?string $downloadUrl = null,
        public ?string $message = null,
        public ?string $error = null
    ) {}

    /**
     * Legacy compatibility - get deliverableId (alias for id)
     */
    public function getDeliverableId(): ?string
    {
        return $this->id;
    }

    /**
     * Create from API response array
     *
     * @param array<string, mixed> $data
     * @return self
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'] ?? null,
            name: $data['name'] ?? null,
            description: $data['description'] ?? null,
            templateId: $data['templateId'] ?? null,
            projectspaceId: $data['projectspaceId'] ?? null,
            deliverableFolderId: $data['deliverableFolderId'] ?? null,
            metadata: $data['metadata'] ?? null,
            createdBy: $data['createdBy'] ?? null,
            orgId: $data['orgId'] ?? null,
            defaultFont: $data['defaultFont'] ?? null,
            createdOn: $data['createdOn'] ?? null,
            updatedOn: $data['updatedOn'] ?? null,
            isActive: $data['isActive'] ?? null,
            fonts: $data['fonts'] ?? null,
            downloadUrl: $data['downloadUrl'] ?? null,
            message: $data['message'] ?? null,
            error: $data['error'] ?? null
        );
    }
}
